drud/ddev
# Platform.sh and generic hosting provider integration, fixes #2384, fixes #2194

## The Problem/Issue/Bug:

Support platform.sh hosting #2384 
User-provided hosting provider integration #2194 

## How this PR Solves The Problem:

New provider technique with a custom yaml file in .ddev/providers

## Manual Testing Instructions:

- [ ] Platform.sh
1. Create an account on platform.sh, get the API token.
2. Follow the instructions in the .ddev/providers/platform.example to set up
3. `ddev pull platform`
4. Verify that the pulled database and files are correct.

- [ ] Flatfile
* Use the flatfile.yaml.example to set up db and files imports
* Try it out

## Automated Testing Overview:

* Added TestPlatformPull
* Added TestFlatfilePull

## Related Issue Link(s):

## Release/Deployment notes:
